pylint/checkers/base/basic_checker.py

```diff
@@ -657,13 +657,54 @@ def _check_misplaced_format_function(self, call_node: nodes.Call) -> None:
             ):
                 self.add_message("misplaced-format-function", node=call_node)
 
+    @staticmethod
+    def _is_terminating_func(node: nodes.Call) -> bool:
+        """Detect call to exit(), quit(), os._exit(), or sys.exit()."""
+        if (
+            not isinstance(node.func, nodes.Attribute)
+            and not (
+                isinstance(node.func, nodes.Name) and node.func.name in {"exit", "quit"}
+            )
+            or isinstance(node.parent, nodes.Lambda)
+        ):
+            return False
+
+        inferred = utils.safe_infer(node.func)
+
+        if (
+            not inferred
+            and isinstance(node.func, nodes.Attribute)
+            and node.func.attrname == "_exit"
+        ):
+            # edge case, try to detect os._exit
+            try:
+                inferred = next(node.func.infer())
+            except (StopIteration, astroid.InferenceError):
+                return False
+            return inferred.qname() == "posix._exit"
+
+        return (
+            inferred.qname() in {"_sitebuiltins.Quitter", "sys.exit"}
+            if inferred
+            else False
+        )
+
     @utils.only_required_for_messages(
-        "eval-used", "exec-used", "bad-reversed-sequence", "misplaced-format-function"
+        "eval-used",
+        "exec-used",
+        "bad-reversed-sequence",
+        "misplaced-format-function",
+        "unreachable",
     )
     def visit_call(self, node: nodes.Call) -> None:
-        """Visit a Call node -> check if this is not a disallowed builtin
-        call and check for * or ** use.
+        """Visit a Call node.
+
+        1. check if this is not a disallowed builtin call
+        2. check for * or ** use.
+        3. check if this call is a terminating function and is followed by unreachable code
         """
```

Comment:
```suggestion
        """Visit a Call node."""
```

This was out of date, we don't do 2 anymore and we do more than 1 + 3. Let's just remove this as it has a high risk of becoming out of date again.